qiita_db/metadata_template.py

```diff
@@ -1146,26 +1146,22 @@ def get_filepaths(self, conn_handler=None):
         return [(fpid, base_fp(fp)) for fpid, fp in filepath_ids]
 
     def categories(self):
-        """Get the categories associated with self
+        """Identifies the metadata columns present in a template
 
         Returns
         -------
-        set
-            The set of categories associated with self
-        """
-        conn_handler = SQLConnectionHandler()
-        table_name = self._table_name(self.study_id)
+        cols : list
+            The static and dynamic category fields
 
-        raw = conn_handler.execute_fetchall("""
-            SELECT column_name
-            FROM information_schema.columns
-            WHERE table_name='{0}'
-                AND table_schema='qiita'""".format(table_name))
+        """
+        cols = get_table_cols(self._table_name(self._id))
+        cols.extend(get_table_cols(self._table)[1:])
 
-        categories = {c[0] for c in raw}
-        categories.remove('sample_id')
+        for idx, c in enumerate(cols):
+            if c in self.translate_cols_dict:
+                cols[idx] = self.translate_cols_dict[c]
 
-        return categories
+        return set(cols)
```

Comment:
the work done in 1160 is not doing anything due to the cast to `set`

the docstring indicates the return is a `list`, however the return is a `set`. It would be great if the return type isn't modified.